shared_model/validators/field_validator.hpp

```diff
@@ -36,39 +36,54 @@ namespace shared_model {
     class FieldValidator {
      public:
       FieldValidator()
-          : account_id_(R"([a-z]{1,9}\@[a-z]{1,9})"),
-            asset_id_(R"([a-z]{1,9}\#[a-z]{1,9})"),
-            ip_address_(
-                "((([0-1]?\\d\\d?)|((2[0-4]\\d)|(25[0-5]))).){3}(([0-1]?\\d\\d?"
-                ")|((2[0-4]\\d)|(25[0-5])))"),
-            name_(R"([a-z]{1,9})"),
-            detail_key_(R"([A-Za-z0-9_]{1,})") {}
+          : account_id_pattern_(R"([a-z]{1,9}\@[a-z]{1,9})"),
```

Comment:
account_id validation has changed to RFC1035.
It is difficult to express it by regex. `boost::spirit::qi`(EBNF) is used and the parser is implemented in `validator::isValidDomainName()`
https://github.com/hyperledger/iroha/pull/694

Reply:
I suggest we make that a separate pull request.